src/sources/azure_blob/queue.rs

```diff
@@ -127,45 +143,76 @@ struct AzureStorageEvent {
+        error: base64::DecodeError
+    },
+
+    #[snafu(display("Failed to utf8 decode message: {}", error))]
+    FailedDecodingUTF8 {
+        error: std::string::FromUtf8Error
+    },
+
+    #[snafu(display("Failed to get blob: {}", error))]
+    FailedToGetBlob {
+        error: azure_core::Error
+    },
+
+    #[snafu(display("Failed to parse {} as subject", subject))]
+    FailedToParseSubject {
+        subject: String,
+    },
+}
+
 async fn proccess_event_grid_message(
     message: Message,
     container_client: &ContainerClient,
     queue_client: &QueueClient,
     bytes_received: Registered<BytesReceived>,
-) -> Option<BlobPack> {
-    let decoded_bytes = match BASE64_STANDARD.decode(&message.message_text) {
-        Ok(decoded) => decoded,
-        Err(e) => {
-            error!("Failed decoding base64: {}", e);
-            return None;
-        }
-    };
-    let decoded_string = match String::from_utf8(decoded_bytes) {
-        Ok(decoded) => decoded,
-        Err(e) => {
-            error!("Failed decoding utf8: {}", e);
-            return None;
-        }
-    };
-    let body: AzureStorageEvent = match serde_json::from_str(decoded_string.as_str()) {
-        Ok(body) => body,
-        Err(e) => {
-            error!("Failed decoding json: {}", e);
-            return None;
-        }
-    };
+) -> Result<Option<BlobPack>, ProcessingError> {
+    let msg_id = message.message_id.clone();
+    let decoded_bytes = BASE64_STANDARD.decode(&message.message_text)
+        .map_err(|e| ProcessingError::FailedDecodingMessageBase64 { error: e })?;
+    let decoded_string = String::from_utf8(decoded_bytes)
+        .map_err(|e| ProcessingError::FailedDecodingUTF8 { error: e })?;
+    let body: AzureStorageEvent = serde_json::from_str(decoded_string.as_str())
+        .map_err(|e| ProcessingError::InvalidQueueMessage {
+            error: e,
+            message_id: msg_id,
+        })?;
     if body.event_type != "Microsoft.Storage.BlobCreated" {
-        trace!(
-            "Ignoring event because of wrong event type: {}",
-            body.event_type
-        );
-        return None;
+        emit!(QueueStorageInvalidEventIgnored{
+           container: container_client.container_name(),
+           subject: &body.subject,
+           event_type: &body.event_type,
+        });
+        return Ok(None);
     }
-    match parse_subject(body.subject) {
+    match parse_subject(body.subject.clone()) {
         Some((container, blob)) => {
             if container != container_client.container_name() {
-                trace!("Container name does not match. Skipping.");
-                return None;
+                return Err(ProcessingError::ContainerNameDoesntMatch {
+                    container: container_client.container_name().to_string(),
+                    message: container,
+                });
             }
             trace!(
                 "New blob created in container '{}': '{}'",
```

Comment:
Please change to sth like: `Detected mew blob created in container.`, so that it's obvious vector doesn't actually create anything.